hw07.c

```diff
@@ -0,0 +1,246 @@
+    self->data[to1D(x,y, tga_height(self))].blue = pix->blue;
+    self->data[to1D(x,y, tga_height(self))].green = pix->green;
+    self->data[to1D(x,y, tga_height(self))].red = pix->red;
+}
+
+void draw_bg(TGAImage *self, RGBA *bg){
+    for (int i = 0; i < tga_height(self); i++)
+    {
+        for (int j = 0; j < tga_width(self); j++)
+        {
+            set_pixel(self, i ,j , bg);
+            printf("bg: i:%d j:%d\n",i ,j);
+        }
+
+    }
+
+}
+
+//pocita indexy nalezejici vykresleni znaku  ty posila do fce ktere zapisuje ty hodnotu do pole clena strukty *data
+void draw_one(TGAImage *self, int dx, int dy, RGBA *pix){
+    for (int i = 0 + dx; i < 195 + dx; i++)
+    {
+        for (int j = 0 + dx; j < 169 + dy; j++)
+        {
+            if (i > 15 + dx && j > 139 + dy)
+            {
+                set_pixel(self, i, j, pix);
+                printf("one: i:%d j:%d\n",i ,j);
+            }
+        }
+    }
+}
+
+void draw_zero(TGAImage *self, int dx, int dy, RGBA *pix){
+    for (int i = 0 + dx; i < 195 + dx; i++)
+    {
+        for (int j = 0 + dx; j < 169 + dy; j++)
+        {
+            if (i > 15 + dx && j > 139 + dy)
+            {
+                set_pixel(self, i, j, pix);
+                printf("one: i:%d j:%d\n",i ,j);
+            }
+        }
+    }
+}
+
+//nevim jestli tohle muze byt jen v mainu nebo pokuud pouziju typedef nejak na tohle pak to jde takhle
+//typedef of array of function pointers (takhle pka func_array je pak data type plus jmesno)
+//typedef TGAImage* (*func_array[])(TGAImage *, size_t, size_t, RGBA *);
+
+void watch_draw_time(TGAImage* self, const int hours, const int minutes){
+
+    FILE *file = fopen("aw.tga","wb");
+    assert(file);
+
+    int firstH = hours / 10;;
+    int secondH = hours % 10;
+    int firstMin = minutes / 10;
+    int secondMin = minutes % 10;
+
+    RGBA fg = {.red = 30, .green = 10, .blue = 10, .alpha = 255};
+//zde musem volat fci s fcnim ukazatel na pole a pdle first/,... hodnot se zavola rovnou dana fce,,nepamatujeu se uz inicializaci atd
+//POLE FUNKCNICHU UKAZATELU
+    void (*func_array[])(TGAImage*, int, int, RGBA*) = {
+        draw_zero, draw_one, /*draw_two, draw_three, draw_foure, draw_five, draw_six, draw seven, draw_eight, draw_nine*/
+    };
+//jen budu omezovat nebo pricit konec zacatek k ouradnicim(prevod do 1D to pak prepocita na ta mista v celem poli, ktere je uz predpripraveno)
+    //draw_one(self,0,0,&fg);
+    /*void (*fptr)(TGAImage*, int, int, RGBA*);
+    fptr = &draw_one;*/
+
+   /* func_array[firstH](self, 0, 0, &fg);
+    func_array[secondH](self, 184, 0, &fg);
+    func_array[firstMin](self, 0, 224, &fg);
+    func_array[secondMin](self, 184, 224, &fg);*/
+
+    int writeCount = fwrite( &self->header, sizeof (TGAheader),1,file); //velikost prvku je header celej a je jen jeden
+    assert(writeCount == 1);
+
+    writeCount = fwrite(&self->data, sizeof(RGBA), tga_width(self) * tga_height(self) * sizeof(RGBA),file);
```

Comment:
to posledni `sizeof(RGBA)` tam nema co delat.

Reply:
odstranil sem